### PR TITLE
feat(java-client): add maven-thrift-plugin

### LINKED ISSUE #1664: Use thrift-maven-plugin to generate thrift sources instead of script
We could use [thrift-maven-plugin](https://mvnrepository.com/artifact/org.apache.thrift/thrift-maven-plugin) to generate thrift sources instead of scripts, just like what parquet has done:

- https://issues.apache.org/jira/browse/PARQUET-1506
- https://github.com/apache/parquet-java/pull/600
- https://github.com/apache/parquet-java/blob/master/parquet-format-structures/pom.xml

### PR DESCRIPTION
**Pull request by shalk**
### What problem does this PR solve? 
fix #1664 

ref: https://issues.apache.org/jira/browse/THRIFT-1536
ref: https://github.com/apache/parquet-mr/pull/600

##### Tests 

- Unit test
- Integration test
- Manual test (add detailed scripts or steps below)
- No code

##### Code changes

- Has exported function/method change
- Has exported variable/fields change
- Has interface methods change
- Has persistent data change

##### Side effects

- Possible performance regression
- Increased code complexity
- Breaking backward compatibility

##### Related changes

- Need to cherry-pick to the release branch
- Need to update the documentation
- Need to be included in the release note

### CODE REVIEW

File: java-client/pom.xml [hunk 1]
```diff
@@ -470,6 +470,36 @@
           </execution>
         </executions>
       </plugin>
+      <plugin>
+        <groupId>org.apache.thrift</groupId>
+        <artifactId>thrift-maven-plugin</artifactId>
+        <version>0.10.0</version>
```
Comment (acelyc111) [inline java-client/pom.xml:476]: For compatibility reason, it's recommend to use 0.11.0 version.

File: java-client/scripts/download_thrift.sh [hunk 2]
```diff
@@ -48,21 +48,4 @@ if ! $thrift -version | grep "0.11.0" ; then
     exit 1
 fi
 
-TMP_DIR=./gen-java
```
Comment (acelyc111) [inline java-client/scripts/download_thrift.sh:51]: It seems `thrift` will not be used any where in java-client, can we remove this file entirely?

Comment (shalk) [reply java-client/scripts/download_thrift.sh:51]: thrift will be used when `mvn compile` or `mvn test`. If i remove the file , if i remove the file  https://github.com/apache/incubator-pegasus/blob/master/.github/workflows/test_python-client.yml#L68 will be add in to github action

File: java-client/pom.xml (hunk 1 above)
Comment (shalk) [reply java-client/pom.xml:476]: This is plugin version, i found this is the latest version. thrift java lib version will be 0.11.0 which defined in libthrift.version

File: java-client/scripts/download_thrift.sh (hunk 2 above)
Comment (acelyc111) [reply java-client/scripts/download_thrift.sh:51]: OK, let's keep it in current step. We can do that in next patches.

**Review by acelyc111 — APPROVED** [review]

**Review by empiredan — APPROVED** [review]

**Pull request merged**